client/src/api/common/data.ts

```diff
@@ -51,7 +44,73 @@ export type ModuleData = Static<typeof ModuleData>;
  * @returns Data
  */
 export function useModuleData(id: number): ModuleData {
-  return usePayload(`module-${id}-data`, ModuleData) ?? { sensors: [] };
+  const [data, setData] = useState<ModuleData>({ sensors: [] });
+
+  // eslint-disable-next-line no-undef
+  useChannelShaped(`wireless_module-${id}-data`, ModuleData, setData);
+
+  return data;
+}
+
+const ModuleBattery = Record({
+  /** Battery percentage */
+  percentage: Number,
+});
+
+type _ModuleBattery = Static<typeof ModuleBattery>;
+
+export interface ModuleBattery extends _ModuleBattery {}
+
+/**
+ * Get battery percentage for wireless module
+ *
+ * @param id ID of module
+ * @returns Data
+ */
+export function useModuleBattery(id: number): ModuleBattery | null {
+  const [battery, setBattery] = useState<ModuleBattery | null>(null);
+
+  useChannelShaped(`wireless_module-${id}-battery`, ModuleBattery, setBattery);
+
+  return battery;
+}
+
+/**
+ * Get status for module
+ *
+ * @param id ID of module
+ * @param name Name used for display
+ * @returns Status
+ */
+export function useModuleStatus(id: number, name: string): WMStatus {
+  // Check if module is online
+  const [online, setOnline] = useState<boolean>(false);
+  useEffect(() => {
+    emit('get-payload', ['wireless_module', `${id}`, 'online']);
+  }, [id]);
+  useChannelShaped(
+    `wireless_module-${id}-online`,
+    Boolean,
+    (data: boolean | null) => setOnline(data ?? false),
+  );
+
+  const data = useModuleData(id).sensors;
+  const batteryPercentage = useModuleBattery(id)?.percentage ?? -1;
+
+  if (!online) {
+    return {
+      moduleName: name,
+      online: false,
+    };
+  }
+
+  return {
+    moduleName: name,
+    online: true,
+    data,
+    batteryPercentage,
+    mqttAddress: '501',
```

Comment:
Maybe something like `Unknown address` would be better, so that users know what the message means.

Reply:
Added description
493ffbc